src/number_theory/primorial.lean

```diff
@@ -0,0 +1,217 @@
+import tactic
+
+open_locale big_operators
+open finset
+open nat
+
+-- This is going into stdlib
+lemma sum_range_choose_halfway (m : ℕ) : ∑ i in range (m + 1), choose (2 * m + 1) i = 4 ^ m :=
+   by sorry
+
+private lemma add_one_zero (a : ℕ) (p : a + 1 = 0) : false := by linarith
+
+private lemma choose_symm_half (m : ℕ) : choose (2 * m + 1) (m + 1) = choose (2 * m + 1) m :=
+by apply choose_symm_of_eq_add; ring
+
+lemma extend_prime_product_odd (n : ℕ) (n_big : 2 < n) (r : n % 2 = 0) :
+  ∏ i in (filter prime (range n)), i = ∏ i in (filter prime (range (n + 1))), i
+  :=
+begin
+  have not_prime : ¬prime n,
+    { intros is_prime,
+      cases (prime.eq_two_or_odd is_prime),
+      { subst h, linarith, },
+      { linarith }, },
+  have t :
+    ∏ i in (filter prime (range (n + 1))), i
+    = (∏ i in (filter prime (range n)), i) * (if prime n then n else 1),
+    { have s : range n.succ = insert n (range n), by exact range_succ,
+      rw [s, filter_insert, if_neg not_prime, if_neg not_prime],
+      ring, },
+  rw t,
+  simp only [not_prime, mul_one, if_false],
+end
+
+private lemma range_contains_upper_bound (n : ℕ) : n ∈ range (n + 1) :=
+by simp only [succ_pos', lt_add_iff_pos_right, mem_range]
+
+private lemma select_from_sum (f : ℕ → ℕ) (s : finset ℕ) (a : ℕ) (t : a ∈ s) : f a ≤ ∑ i in s, f i :=
+by rw <- finset.insert_erase t; simp
+
+lemma bound_binomial_coefficient (n : ℕ) : choose (2 * n + 1) n ≤ 4 ^ n :=
+begin
+  have t : choose (2 * n + 1) n ≤ ∑ i in range (n + 1), choose (2 * n + 1) i,
+    exact select_from_sum (choose (2 * n + 1)) (range (n + 1)) n (range_contains_upper_bound n),
+  simpa only [sum_range_choose_halfway n] using t
+end
+
+private lemma can_halve : ∀ (m : ℕ), (m % 2 = 0) → ∃ n, 2 * n = m
```

Comment:
Have you looked at stuff in the library tagged with `parity_simps`? There's probably a bunch of useful lemmas among that (or close by).